Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -1671,7 +1671,30 @@ - (void)endEditingAndUpdateFocusAppearanceWithReason:(EndEditingReason)reason
     }
 
     [self _cancelInteraction];
-    [_textInteractionAssistant deactivateSelection];
+
+    BOOL shouldDeactivateSelection = [&] {
+#if PLATFORM(MACCATALYST)
+        if (reason == EndEditingReasonResigningFirstResponder) {
+            // This logic is based on a similar check on macOS (in WebViewImpl::resignFirstResponder()) to
+            // maintain the active selection when resigning first responder, if the new responder is in a
+            // modal popover or panel.
+            // FIXME: Ideally, this would additionally check that the new first responder corresponds to
+            // a view underneath this view controller; however, there doesn't seem to be any way of doing
+            // so at the moment. In lieu of this, we can at least check that the web view itself isn't
+            // inside the popover.
+            auto *controller = [UIViewController _viewControllerForFullScreenPresentationFromView:self];
+            for (UIView *view = _webView.get().get(); view; view = view.superview) {
+                if (view == controller.view)
+                    return YES;
+            }
```

Comment:
Good catch! Changed to use `-isDescendantOfView:`.

I also changed the call to `-view` to `-viewIfLoaded` instead, so that this won't ever force a view controller to initialize its view.